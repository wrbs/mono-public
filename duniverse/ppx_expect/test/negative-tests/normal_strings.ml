let%expect_test "short quoted string" =
  print_string "foo\nbar\n";
  [%expect ""]
;;

let%expect_test "long quoted string" =
  print_string
    {|
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXX
    |};
  [%expect ""]
;;

let%expect_test "quoted strings with leading spaces" =
  print_string {|
    live
      long
        and
    prosper
    |};
  [%expect ""];
  print_string
    {|
    live
      loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong
        and
    prosper
    |};
  [%expect ""]
;;
