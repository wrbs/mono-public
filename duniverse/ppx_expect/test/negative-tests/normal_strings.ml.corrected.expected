let%expect_test "short quoted string" =
  print_string "foo\nbar\n";
  [%expect " \n foo\n bar\n "]
;;

let%expect_test "long quoted string" =
  print_string
    {|
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    XXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXX
    |};
  [%expect
    " \n\
    \ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n\
    \ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n\
    \ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n\
    \ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX \
     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n\
    \ XXXXXXXXXXXXXXXXXXXX XXXXXXXXXXXXXXXXXXX\n\
    \ "]
;;

let%expect_test "quoted strings with leading spaces" =
  print_string
    {|
    live
      long
        and
    prosper
    |};
  [%expect " \n live\n   long\n     and\n prosper\n "];
  print_string
    {|
    live
      loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong
        and
    prosper
    |};
  [%expect
    " \n\
    \ live\n\
    \   loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong\n\
    \     and\n\
    \ prosper\n\
    \ "]
;;
